packages/runtime/container-runtime/src/orderedClientElection.ts

```diff
@@ -110,122 +146,260 @@ export class OrderedClientElection extends TypedEventEmitter<IOrderedClientElect
+    /** Event fires when the currently elected client changes. */
+    (event: "election", listener: (
+        /** Newly elected client. */
+        client: ITrackedClient | undefined,
+        /** Sequence number where election took place. */
+        sequenceNumber: number,
+        /** Previously elected client. */
+        prevClient: ITrackedClient | undefined,
+    ) => void);
+}
+
+/** Serialized state of IOrderedClientElection. */
+export interface ISerializedElection {
+    /** Sequence number at the time of the latest election. */
+    readonly electionSequenceNumber: number;
+    /** Most recently elected client id. */
+    readonly electedClientId: string | undefined;
+}
+
+/** Contract for maintaining a deterministic client election based on eligibility. */
+export interface IOrderedClientElection extends IEventProvider<IOrderedClientElectionEvents> {
+    /** Count of eligible clients in the collection. */
+    readonly eligibleCount: number;
+    /** Currently elected client. */
+    readonly electedClient: ITrackedClient | undefined;
+    /** Sequence number of most recent election. */
+    readonly electionSequenceNumber: number;
+    /** Marks the currently elected client as invalid, and elects the next eligible client. */
+    incrementElectedClient(sequenceNumber: number): void;
+    /** Resets the currently elected client back to the oldest eligible client. */
+    resetElectedClient(sequenceNumber: number): void;
+    /** Peeks at what the next elected client would be if incrementElectedClient were called. */
+    peekNextElectedClient(): ITrackedClient | undefined;
+    /** Returns a sorted array of all the eligible clients in the collection. */
+    getAllEligibleClients(): ITrackedClient[];
+    /** Serialize election data */
+    serialize(): ISerializedElection;
+}
+
+/**
+ * Adapter for OrderedClientCollection, with the purpose of deterministically maintaining
+ * a currently elected client, excluding ineligible clients, in a distributed fashion.
+ * This can be true as long as incrementElectedClient and resetElectedClient calls
+ * are called under the same conditions for all clients.
+ */
+export class OrderedClientElection
+    extends TypedEventEmitter<IOrderedClientElectionEvents>
+    implements IOrderedClientElection {
+    private _eligibleCount: number = 0;
+    private _electedClient: ILinkedClient | undefined;
+    private _electionSequenceNumber: number;
+
+    public get eligibleCount() {
+        return this._eligibleCount;
+    }
+    public get electedClient() {
+        return this._electedClient;
+    }
+    public get electionSequenceNumber() {
+        return this._electionSequenceNumber;
+    }
+
+    constructor(
+        logger: ITelemetryLogger,
+        private readonly orderedClientCollection: IOrderedClientCollection,
+        /** Serialized state from summary or current sequence number at time of load if new. */
+        initialState: ISerializedElection | number,
```

Comment:
>ISerializedElection | number [](http://example.com/codeflow?start=22&length=28)

why do we need to support number only?

Reply:
When this is first rolled out we won't have an initial state to load from. So we pass the initial sequence number as a baseline.